extensions/applicationinsights-clickanalytics-js/src/DataCollector.ts

```diff
@@ -141,7 +143,9 @@ export function sanitizeUrl(config: IClickAnalyticsConfiguration, location: Loca
     if (!!config.urlCollectQuery) { // false by default
         url += (isValueAssigned(location.search)? location.search : "");
     }
-
+    if (rootConfig) {
```

Comment:
Instead of checking config everywhere you call fieldRedaction, add a config validation inside that method if is not already there